src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java

```diff
@@ -110,7 +113,103 @@ public void putLoanPolicyStorageLoanPoliciesByLoanPolicyId(
+    Errors errors = ValidationHelper.createValidationErrorMessage(field, "" + value, message);
+    Response response = Response.status(422).header("Content-Type", "application/json").entity(errors).build();
+    asyncResultHandler.handle(Future.succeededFuture(response));
+  }
+
+  /**
+   * @return true if bool is TRUE, false if bool is FALSE or null
+   */
+  private boolean is(Boolean bool) {
+    return Boolean.TRUE.equals(bool);
+  }
+
+  void validate(LoanPolicy loanPolicy, Handler<AsyncResult<Response>> asyncResultHandler, Runnable runIfValid) {
+    final boolean isFixed = loanPolicy.getLoansPolicy() != null &&
+                            loanPolicy.getLoansPolicy().getProfileId() != null &&
+                            loanPolicy.getLoansPolicy().getProfileId().equalsIgnoreCase("Fixed");
+
+    // alternate fixed due date
+    if ((is(loanPolicy.getRenewable()) &&
+         loanPolicy.getRenewalsPolicy() != null &&
+         is(loanPolicy.getRenewalsPolicy().getDifferentPeriod()) &&
+         isFixed &&
+         loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId() == null
+        )
+        ||
+        ( loanPolicy.getRenewalsPolicy() != null
+          &&
+          ( ! is(loanPolicy.getRenewable()) ||
+            ! is(loanPolicy.getRenewalsPolicy().getDifferentPeriod())
+          )
+          && loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId() != null
+       )) {
+      String message =
+          "Alternate fixed due date cannot be " + loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId() +
+          " if renewable is " + loanPolicy.getRenewable() +
+          ", different period is " + loanPolicy.getRenewalsPolicy().getDifferentPeriod() +
+          " and profile is " + loanPolicy.getLoansPolicy().getProfileId();
+      respond422("alternateFixedDueDateScheduleId",
+          loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId(),
+          message, asyncResultHandler);
+      return;
+    }
+
+    // fixed profile id
+    if ( (is(loanPolicy.getLoanable()) &&
+          isFixed &&
+          loanPolicy.getLoansPolicy().getFixedDueDateScheduleId() == null
+         )
+         ||
+         // TODO: consider adjusting the message
+         (! is(loanPolicy.getLoanable()) &&
+          loanPolicy.getLoansPolicy() != null &&
+          // TODO: consider removing this last condition
+          loanPolicy.getLoansPolicy().getFixedDueDateScheduleId() == null
+         )
+        ) {
+      String message = "Fixed due date cannot be null if loanable is " +
+        loanPolicy.getLoanable() + " and profile is of type fixed";
+      respond422("fixedDueDateScheduleId",
+          loanPolicy.getLoansPolicy().getFixedDueDateScheduleId(),
+          message, asyncResultHandler);
+      return;
+    }
+
+    // alternate renewal loan period
+    if (isFixed
+        && is(loanPolicy.getRenewable())
```

Comment:
The formatting of these large conditionals is inconsistent, please pick a format and use it throughout

Reply:
The formatting is exactly the formatting of the *.drl files that are to be deleted.

Reply:
The formatting is kept to make code review easy.

Reply:
Personally, from my reviewing perspective, I'm not comparing this code to the deleted drools files, I'm considering this as new code. I don't think the formatting needs to be retained.